### PR TITLE
[Terraform] Add support for dns_cache_config to resource_container_cluster

### PR DESCRIPTION
**Pull request by mcfedr**
Fixes https://github.com/terraform-providers/terraform-provider-google/issues/5150

Adds support for NodeLocal DNSCache addon to terraform.

```release-note:enhancement
container: added `dns_cache_config` field to `google_container_cluster` resource
```

### CODE REVIEW

File: third_party/terraform/resources/resource_container_cluster.go.erb [hunk 1]
```diff
@@ -2617,6 +2642,14 @@ func flattenClusterAddonsConfig(c *containerBeta.AddonsConfig) []map[string]inte
 			},
 		}
 	}
+
+	if c.CloudRunConfig != nil {
```
Comment (chrisst) [inline third_party/terraform/resources/resource_container_cluster.go.erb]: I think you meant `DnsCacheConfig`

Comment (mcfedr) [reply third_party/terraform/resources/resource_container_cluster.go.erb]: ah, never copy paste :)

**Comment by chrisst** [comment]
I just ran the tests and they are failing because the default version of gke doesn't support this new addon. I'll bump the version in the tests and push a commit to your branch if you have allowed edits.

**Comment by chrisst** [comment]
Looks like I don't have permission. Can you update the PR with something similar to the following:
```
diff --git a/third_party/terraform/tests/resource_container_cluster_test.go.erb b/third_party/terraform/tests/resource_container_cluster_test.go.erb
 index 2ce8720e3..37404dfdb 100644
 --- a/third_party/terraform/tests/resource_container_cluster_test.go.erb
 +++ b/third_party/terraform/tests/resource_container_cluster_test.go.erb
 @@ -146,6 +146,7 @@ func TestAccContainerCluster_withAddons(t *testing.T) {
                 ResourceName:        "google_container_cluster.primary",
                 ImportState:         true,
                 ImportStateVerify:   true,
 +               ImportStateVerifyIgnore: []string{"min_master_version"},
             },
             {
                 Config: testAccContainerCluster_updateAddons(clusterName),
 @@ -154,6 +155,7 @@ func TestAccContainerCluster_withAddons(t *testing.T) {
                 ResourceName:        "google_container_cluster.primary",
                 ImportState:         true,
                 ImportStateVerify:   true,
 +               ImportStateVerifyIgnore: []string{"min_master_version"},
             },
         },
     })
 @@ -1957,6 +1959,8 @@ resource "google_container_cluster" "primary" {
    location           = "us-central1-a"
    initial_node_count = 1

 +  min_master_version = "latest"
 +
    addons_config {
      http_load_balancing {
        disabled = true
 @@ -1991,6 +1995,8 @@ resource "google_container_cluster" "primary" {
    location           = "us-central1-a"
    initial_node_count = 1

 +  min_master_version = "latest"
 +
    addons_config {
      http_load_balancing {
        disabled = false
```

**Comment by mcfedr** [comment]
I have got "Allow edits from maintainers. " ticked, but i guess you dont fall in that group :)

I've applied the patch.

**Pull request merged**